packages/NetteToSymfony/src/Rector/ClassMethod/RenameEventNamesInEventSubscriberRector.php

```diff
@@ -0,0 +1,197 @@
+        'nette.application.error' => ['Symfony\Component\HttpKernel\KernelEvents', 'EXCEPTION'],
+    ];
+
+    /**
+     * @var string[][]
+     */
+    private $netteClassConstToSymfonyClassConstMap = [
+        'Contributte\Events\Extra\Event\Application::NAME' => ['Symfony\Component\HttpKernel\KernelEvents', 'REQUEST'],
+        'Contributte\Events\Extra\Event\ShutdownEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'TERMINATE',
+        ],
+        'Contributte\Events\Extra\Event\RequestEvent::NAME' => ['Symfony\Component\HttpKernel\KernelEvents', 'REQUEST'],
+        'Contributte\Events\Extra\Event\PresenterEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\PresenterStartupEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\PresenterShutdownEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\ResponseEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'RESPONSE',
+        ],
+        'Contributte\Events\Extra\Event\ErrorEvent::NAME' => ['Symfony\Component\HttpKernel\KernelEvents', 'EXCEPTION'],
+    ];
+
+    /**
+     * @var BetterNodeFinder
+     */
+    private $betterNodeFinder;
+
+    public function __construct(BetterNodeFinder $betterNodeFinder)
+    {
+        $this->betterNodeFinder = $betterNodeFinder;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Changes event names from Nette ones to Symfony ones', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class SomeClass implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return ['nette.application' => 'someMethod'];
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class SomeClass implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [\SymfonyEvents::KERNEL => 'someMethod'];
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $classNode = $node->getAttribute(Attribute::CLASS_NODE);
+        if ($classNode === null) {
+            return null;
+        }
+
+        if (! $this->isType($classNode, 'Symfony\Component\EventDispatcher\EventSubscriberInterface')) {
```

Comment:
Just to make sure, does this cover subtypes of this interface? `Interface XyzShit extends EventSubscriberInterface`?

Reply:
Yes yes, it does check type of the node. Not just strings